pytensor/link/c/exceptions.py

```diff
@@ -1,4 +1,7 @@
-from setuptools.errors import CompileError as BaseCompileError
+try:
```

Comment:
Can you add a comment?

Reply:
I want to add a warning here. I'm tracking down the various relevant setuptools versions to cite.

Reply:
@ferrine, @twiecki, I added a warning. Does this look helpful?

```
In [1]: import pytensor
~/repos/pytensor/pytensor/link/c/exceptions.py:14: UserWarning: You appear to be using an ancient
version of setuptools: v57.4.0. Please upgrade to at least v59.0.0. Support for this version of
setuptools is provisionary and may be removed without warning in the future.
  warnings.warn(

In [2]:
```
I'm not sure what the dangling `warnings.warn(` is about, but I'm not sure we should care.